.github/pull_request_template.md

```diff
@@ -0,0 +1,43 @@
+#### What does the PR do?
+<!-- Describe your pull request here. Please read the text below the line, and make sure you follow the checklist.-->
+
+
+#### Checklist
+- [ ] Populated all fields in this template.
+- [ ] The PR title reflects the change.
+- [ ] Changes are described in the pull request, or an [existing issue is referenced](https://github.com/triton-inference-server/server/issues).
+- [ ] Added [test plan](#test-plan) and verified test passes.
+- [ ] Verified that the PR passes existing CI.
+- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+- [ ] CodeQL suggestions resolved
+- [ ] Lint Gods are pleased (i.e. pre-commit checks are green)
+- [ ] Optional: Additional screenshots for behavior/output changes with before/after.
+
+#### Type:
+- [ ] Feature
+- [ ] Bugfix
+- [ ] Enhancement
+- [ ] Chore / doc / build etc
+- [ ] Other (please specify)
+
+#### Related PRs:
+<!-- Related PRs from other Repositories -->
+
+#### Where should the reviewer start?
+<!-- call out specific files that should be looked at closely -->
+
+#### Test plan:
+<!-- list steps to verify -->
```

Comment:
Is pointing to the test sufficient here - or is there a different format for test plan?